src/connector/src/sink/snowflake_redshift/snowflake.rs

```diff
@@ -0,0 +1,1115 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::num::NonZeroU64;
+use std::collections::BTreeMap;
+
+use anyhow::anyhow;
+use phf::{Set, phf_set};
+use risingwave_common::array::StreamChunk;
+use risingwave_common::catalog::{ColumnDesc, ColumnId, Field, Schema};
+use risingwave_common::types::DataType;
+use risingwave_pb::connector_service::{SinkMetadata, sink_metadata};
+use sea_orm::DatabaseConnection;
+use serde::Deserialize;
+use serde_with::{DisplayFromStr, serde_as};
+use thiserror_ext::AsReport;
+use tokio::sync::mpsc::UnboundedSender;
+use tonic::async_trait;
+use with_options::WithOptions;
+
+use crate::connector_common::IcebergSinkCompactionUpdate;
+use crate::enforce_secret::EnforceSecret;
+use crate::sink::coordinate::CoordinatedLogSinker;
+use crate::sink::decouple_checkpoint_log_sink::default_commit_checkpoint_interval;
+use crate::sink::file_sink::s3::S3Common;
+use crate::sink::jdbc_jni_client::{self, JdbcJniClient};
+use crate::sink::remote::CoordinatedRemoteSinkWriter;
+use crate::sink::snowflake_redshift::{AugmentedChunk, SnowflakeRedshiftSinkS3Writer};
+use crate::sink::writer::SinkWriter;
+use crate::sink::{
+    Result, SINK_TYPE_APPEND_ONLY, SINK_TYPE_OPTION, SINK_TYPE_UPSERT, Sink, SinkCommitCoordinator,
+    SinkCommittedEpochSubscriber, SinkError, SinkParam, SinkWriterMetrics, SinkWriterParam,
+};
+
+pub const SNOWFLAKE_SINK: &str = "snowflake";
+pub const SNOWFLAKE_SINK_ROW_ID: &str = "__row_id";
+pub const SNOWFLAKE_SINK_OP: &str = "__op";
+
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, WithOptions)]
+pub struct SnowflakeConfig {
+    #[serde(rename = "type")]
+    pub r#type: String,
+
+    #[serde(rename = "intermediate.table.name")]
+    pub snowflake_cdc_table_name: Option<String>,
+
+    #[serde(rename = "target.table.name")]
```

Comment:
Since iceberg call this `table.name`, we can make it consistent here and call it `table.name` as well.